indexer-api/src/infra/storage/dust.rs

```diff
@@ -0,0 +1,130 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{
+    domain::{dust::DustGenerationStatus, storage::dust::DustStorage},
+    infra::storage::Storage,
+};
+use fastrace::trace;
+use indexer_common::{domain::CardanoStakeKey, infra::sqlx::U128BeBytes};
+use indoc::indoc;
+
+impl DustStorage for Storage {
+    #[trace]
+    async fn get_dust_generation_status(
+        &self,
+        cardano_stake_keys: &[CardanoStakeKey],
+    ) -> Result<Vec<DustGenerationStatus>, sqlx::Error> {
+        let mut statuses = vec![];
+
+        for stake_key in cardano_stake_keys {
+            // Query registration info.
+            let registration_query = indoc! {"
+                SELECT dust_address, valid
+                FROM cnight_registrations
+                WHERE cardano_address = $1 AND removed_at IS NULL
+                ORDER BY registered_at DESC
+                LIMIT 1
+            "};
+
+            let result = sqlx::query_as::<_, (Vec<u8>, bool)>(registration_query)
+                .bind(stake_key.as_ref())
+                .fetch_optional(&*self.pool)
+                .await?;
+
+            let (dust_address, registered) = match result {
+                Some((addr, valid)) => {
+                    let address_array: [u8; 32] = addr
+                        .try_into()
+                        .map_err(|_| sqlx::Error::Decode("Invalid DUST address length".into()))?;
+                    (
+                        indexer_common::domain::DustAddress::from(address_array),
+                        valid,
+                    )
+                }
+                None => (indexer_common::domain::DustAddress::from([0u8; 32]), false),
+            };
+
+            let mut generation_rate = 0u128;
+            let mut current_capacity = 0u128;
+            let mut night_balance = 0u128;
+
+            // Query active generation info if registered.
+            if registered {
+                let generation_query = indoc! {"
+                    SELECT value, ctime
+                    FROM dust_generation_info
+                    WHERE owner = $1 AND dtime IS NULL
+                    ORDER BY ctime DESC
+                    LIMIT 1
+                "};
+
+                let result = sqlx::query_as::<_, (U128BeBytes, i64)>(generation_query)
+                    .bind(dust_address.as_ref())
+                    .fetch_optional(&*self.pool)
+                    .await?;
+
+                if let Some((value, ctime)) = result {
+                    let value_u128: u128 = value.into();
```

Comment:
Do not add type names to identifiers. 
Replace with: `let value: u128 = value.into()`.
Or better: `let value = u128::from(value)`